sdks/python/apache_beam/coders/typecoders.py

```diff
@@ -116,10 +116,18 @@ def register_coder(self, typehint_type, typehint_coder_class):
           'Received %r instead.' % typehint_coder_class)
     if typehint_type not in self.custom_types:
       self.custom_types.append(typehint_type)
+    if typehint_type.__module__ == '__main__':
```

Comment:
Seems to be failures if `typehint_type` is None
```suggestion
    if typehint and typehint_type.__module__ == '__main__':
```